packages/functions/src/oidc/get-vercel-oidc-token.ts

```diff
@@ -29,9 +30,25 @@ export async function getVercelOidcToken(): Promise<string> {
 
   if (!token) {
     throw new Error(
-      `The 'x-vercel-oidc-token' header is missing from the request. Do you have the OIDC option enabled in the Vercel project settings?`
+      `The Vercel OIDC token is missing in your environment. Do you have the OIDC option enabled in the Vercel project settings?`
     );
   }
 
+  if (process.env.NODE_ENV !== 'production') {
+    const { createRemoteJWKSet, jwtVerify, errors } = await import('jose');
+    try {
+      const jwks = createRemoteJWKSet(new URL('https://oidc.vercel.com'));
+      await jwtVerify(token, jwks);
+    } catch (error) {
+      if (error instanceof errors.JOSEError) {
+        console.warn(
+          'Invalid Vercel OIDC Token. Do you have multiple `.env.*` files with `VERCEL_OIDC_TOKEN` defined?',
+          error
+        );
```

Comment:
nit: should we have a custom error class instead of printing a warning? This will help having a nice `@throws` and allowing people to keep, ignore, reroute the warnings etc